{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "d8bfa5d5-a96c-494b-b58a-5b6a932e6ff9",
   "metadata": {},
   "source": [
    "### Research question 2: Do protein modifications differ between SPP vs. NAT?\n",
    "\n",
    "#### Protein Modifications\n",
    "- Proteins are the building blocks of cells and play crucial roles in various biological processes. Protein modifications refer to the chemical alterations that occur to proteins after they are synthesized. These modifications can occur through several mechanisms, including enzymatic reactions, post-translational modifications (PTMs), and interactions with other molecules.\n",
    "- Common types of protein modifications include phosphorylation, acetylation, methylation, glycosylation, ubiquitination, and many others. These modifications can alter a protein's structure, stability, localization, activity, and interactions with other molecules. Understanding protein modifications is essential in deciphering cellular processes, disease mechanisms, and potential therapeutic targets, such as in malaria research.\n",
    "\n",
    "#### Plasmodium proteins signal peptide peptidase (SPP):\n",
    "- Plasmodium proteins signal peptide peptidase, or SPP, is an enzyme found in the Plasmodium parasite, which causes malaria. Its main job is to process and modify proteins that are important for the parasite's survival and ability to infect human cells.\n",
    "- When proteins are made in cells, they usually have a \"signal peptide\" attached to them. This signal peptide acts like a tag, telling the cell where the protein needs to go. However, before the protein can reach its destination, the signal peptide needs to be removed. That's where SPP comes in.\n",
    "- SPP is responsible for cutting off the signal peptide from the protein. It acts like a pair of molecular scissors, recognizing the specific region of the protein where the signal peptide is attached and snipping it off. Once the signal peptide is removed, the protein can then be transported to its proper location within the Plasmodium parasite.\n",
    "- The role of SPP in Plasmodium is important because it ensures that proteins are correctly processed and directed to the right places. This helps the parasite survive and carry out its infectious activities. Researchers studying malaria may be interested in understanding SPP's function and exploring its potential as a target for developing new treatments against the disease.\n",
    "\n",
    "#### N-Acetyltransferase (NAT)\n",
    "- N-Acetyltransferases (NATs) are a class of enzymes that catalyze the transfer of an acetyl group from acetyl-coenzyme A (acetyl-CoA) to the amino group of various substrates, including small molecules, drugs, and proteins. NATs are involved in the process of acetylation, one of the common post-translational modifications of proteins.\n",
    "- In the context of malaria research, NATs can be of interest due to their potential involvement in the modification of proteins within the Plasmodium parasite or host cells. The acetylation of proteins mediated by NATs can affect protein stability, function, and protein-protein interactions, thereby influencing various cellular processes. By studying the role of NATs in malaria, researchers can gain insights into the molecular mechanisms underlying parasite-host interactions and identify potential targets for intervention."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "025fb736-7002-4f9e-98e4-5c9cd6d31849",
   "metadata": {
    "pycharm": {
     "is_executing": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 19264 entries, 0 to 19263\n",
      "Data columns (total 20 columns):\n",
      " #   Column                                   Non-Null Count  Dtype  \n",
      "---  ------                                   --------------  -----  \n",
      " 0   Experiment name                          19264 non-null  object \n",
      " 1   Biological sample category               19264 non-null  object \n",
      " 2   Protein group                            19264 non-null  object \n",
      " 3   Protein accession number                 19264 non-null  object \n",
      " 4   Protein name                             19264 non-null  object \n",
      " 5   Protein identification probability       19264 non-null  float64\n",
      " 6   Protein percentage of total spectra      19264 non-null  float64\n",
      " 7   Number of unique peptides                19264 non-null  int64  \n",
      " 8   Number of unique spectra                 19264 non-null  int64  \n",
      " 9   Number of total spectra                  19264 non-null  int64  \n",
      " 10  Peptide sequence                         19264 non-null  object \n",
      " 11  Previous amino acid                      19264 non-null  object \n",
      " 12  Next amino acid                          19264 non-null  object \n",
      " 13  Peptide identification probability       19264 non-null  float64\n",
      " 14  Modifications identified by spectrum     6672 non-null   object \n",
      " 15  Peptide start index                      19264 non-null  int64  \n",
      " 16  Peptide stop index                       19264 non-null  int64  \n",
      " 17  Unnamed: 17                              13689 non-null  float64\n",
      " 18  Normalized protein percentage            19264 non-null  float64\n",
      " 19  Corrected normalized protein percentage  19264 non-null  float64\n",
      "dtypes: float64(6), int64(5), object(9)\n",
      "memory usage: 2.9+ MB\n"
     ]
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import os\n",
    "\n",
    "pd.set_option('display.max_colwidth', None)\n",
    "\n",
    "data = pd.read_excel('Malaria_Research_Data.xlsx', header=0)\n",
    "total_spectra = data.groupby('Biological sample category')['Protein percentage of total spectra'].sum()\n",
    "\n",
    "# Normalize the protein percentages within each group\n",
    "data['Normalized protein percentage'] = data.groupby('Biological sample category')['Protein percentage of total spectra'].transform(lambda x: (x / x.sum()) * 100)\n",
    "\n",
    "# Calculate the corrected normalized protein percentage\n",
    "data['Corrected normalized protein percentage'] = data.groupby('Biological sample category')['Normalized protein percentage'].transform(lambda x: (x / x.sum()) * 100)\n",
    "\n",
    "data.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "6264bf20-b640-4e57-8565-271e9f5accbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# collect protein modification columns from source dataframe\n",
    "# protein_mods_data = data[['Biological sample category', 'Protein accession number', 'Peptide sequence', 'Peptide identification probability', 'Modifications identified by spectrum']]\n",
    "# print(f\"\\nprotein_mods_data dataset info:\\n\")\n",
    "# print(protein_mods_data.info())\n",
    "\n",
    "# create subset of protein_mods for SPP\n",
    "protein_mods_spp = data[protein_mods_data['Biological sample category'] == 'SPP']\n",
    "\n",
    "# create subset of protein_mods for NAT\n",
    "protein_mods_nat = data[protein_mods_data['Biological sample category'] == 'NAT']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "fbc202ae-4db6-48e5-8197-9371ca4298f7",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The total number of modificaitons for SPP: 3341\n",
      "\n",
      "The number of SPP modfications identified by spectrum:\n",
      "\n",
      "Modifications identified by spectrum\n",
      "Carbamidomethyl (+57)                                                                   1039\n",
      "Deamidated (+1)                                                                          922\n",
      "Acetyl (+42)                                                                              77\n",
      "Deamidated (+1), Carbamidomethyl (+57)                                                    39\n",
      "Oxidation (+16)                                                                          549\n",
      "Oxidation (+16), Acetyl (+42)                                                             29\n",
      "Carbamidomethyl (+57), Carbamidomethyl (+57)                                             172\n",
      "Deamidated (+1), Deamidated (+1)                                                          94\n",
      "Carbamidomethyl (+57), Deamidated (+1)                                                    60\n",
      "Acetyl (+42), Deamidated (+1)                                                             59\n",
      "Deamidated (+1), Deamidated (+1), Deamidated (+1)                                          8\n",
      "Carbamidomethyl (+57), Deamidated (+1), Carbamidomethyl (+57)                              5\n",
      "Oxidation (+16), Carbamidomethyl (+57)                                                    50\n",
      "Deamidated (+1), Carbamidomethyl (+57), Carbamidomethyl (+57)                             10\n",
      "Deamidated (+1), Oxidation (+16)                                                          21\n",
      "Carbamidomethyl (+57), Oxidation (+16)                                                    46\n",
      "Carbamidomethyl (+57), Carbamidomethyl (+57), Oxidation (+16)                              3\n",
      "Oxidation (+16), Deamidated (+1), Deamidated (+1)                                          1\n",
      "Deamidated (+1), Deamidated (+1), Deamidated (+1), Deamidated (+1)                         2\n",
      "Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57)                        5\n",
      "Oxidation (+16), Deamidated (+1)                                                          36\n",
      "Oxidation (+16), Oxidation (+16)                                                          45\n",
      "Oxidation (+16), Oxidation (+16), Deamidated (+1)                                          9\n",
      "Acetyl (+42), Carbamidomethyl (+57)                                                        4\n",
      "Acetyl (+42), Oxidation (+16)                                                              4\n",
      "Oxidation (+16), Carbamidomethyl (+57), Deamidated (+1), Carbamidomethyl (+57)             1\n",
      "Oxidation (+16), Carbamidomethyl (+57), Carbamidomethyl (+57)                              3\n",
      "Deamidated (+1), Carbamidomethyl (+57), Deamidated (+1)                                    4\n",
      "Deamidated (+1), Deamidated (+1), Carbamidomethyl (+57), Deamidated (+1)                   1\n",
      "Deamidated (+1), Deamidated (+1), Oxidation (+16)                                          5\n",
      "Carbamidomethyl (+57), Deamidated (+1), Carbamidomethyl (+57), Carbamidomethyl (+57)       1\n",
      "Oxidation (+16), Oxidation (+16), Oxidation (+16)                                          1\n",
      "Carbamidomethyl (+57), Deamidated (+1), Deamidated (+1)                                    2\n",
      "Deamidated (+1), Carbamidomethyl (+57), Deamidated (+1), Carbamidomethyl (+57)             1\n",
      "Deamidated (+1), Oxidation (+16), Carbamidomethyl (+57)                                    2\n",
      "Acetyl (+42), Deamidated (+1), Deamidated (+1)                                             9\n",
      "Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)                              3\n",
      "Deamidated (+1), Carbamidomethyl (+57), Oxidation (+16)                                    1\n",
      "Oxidation (+16), Carbamidomethyl (+57), Deamidated (+1)                                    3\n",
      "Oxidation (+16), Acetyl (+42), Deamidated (+1)                                             2\n",
      "Oxidation (+16), Deamidated (+1), Carbamidomethyl (+57)                                    1\n",
      "Deamidated (+1), Deamidated (+1), Carbamidomethyl (+57)                                    1\n",
      "Oxidation (+16), Deamidated (+1), Oxidation (+16)                                          5\n",
      "Acetyl (+42), Carbamidomethyl (+57), Oxidation (+16)                                       2\n",
      "Carbamidomethyl (+57), Oxidation (+16), Carbamidomethyl (+57)                              2\n",
      "Deamidated (+1), Deamidated (+1), Deamidated (+1), Oxidation (+16)                         1\n",
      "Oxidation (+16), Carbamidomethyl (+57), Oxidation (+16)                                    1\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# get count of each distinct occurance of `Modifications identified by spectrum` for SPP\n",
    "spp_modifications = protein_mods_spp['Modifications identified by spectrum']\n",
    "print(f\"The total number of modificaitons for SPP:\", spp_modifications.count())\n",
    "print(f\"\\nThe number of SPP modfications identified by spectrum:\\n\")\n",
    "print(spp_modifications.value_counts(sort = False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b02a10e6-3b60-4772-942f-ce5103f0e03c",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The total number of modificaitons for NAT: 3331\n",
      "\n",
      "The number of NAT modfications identified by spectrum:\n",
      "\n",
      "Modifications identified by spectrum\n",
      "Carbamidomethyl (+57), Carbamidomethyl (+57)                                                                   147\n",
      "Carbamidomethyl (+57)                                                                                          971\n",
      "Acetyl (+42)                                                                                                    90\n",
      "Oxidation (+16), Acetyl (+42)                                                                                   19\n",
      "Deamidated (+1)                                                                                                979\n",
      "Deamidated (+1), Carbamidomethyl (+57)                                                                          44\n",
      "Deamidated (+1), Deamidated (+1)                                                                               103\n",
      "Oxidation (+16), Carbamidomethyl (+57), Deamidated (+1)                                                         10\n",
      "Carbamidomethyl (+57), Deamidated (+1)                                                                          63\n",
      "Oxidation (+16)                                                                                                567\n",
      "Carbamidomethyl (+57), Oxidation (+16)                                                                          21\n",
      "Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57)                                              6\n",
      "Deamidated (+1), Deamidated (+1), Deamidated (+1)                                                               16\n",
      "Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)                                                    4\n",
      "Acetyl (+42), Deamidated (+1)                                                                                   38\n",
      "Oxidation (+16), Deamidated (+1)                                                                                44\n",
      "Oxidation (+16), Oxidation (+16)                                                                                47\n",
      "Oxidation (+16), Carbamidomethyl (+57)                                                                          41\n",
      "Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)                             5\n",
      "Oxidation (+16), Deamidated (+1), Oxidation (+16), Deamidated (+1)                                               1\n",
      "Deamidated (+1), Oxidation (+16)                                                                                30\n",
      "Deamidated (+1), Carbamidomethyl (+57), Deamidated (+1)                                                          3\n",
      "Deamidated (+1), Oxidation (+16), Deamidated (+1)                                                                3\n",
      "Carbamidomethyl (+57), Deamidated (+1), Deamidated (+1), Deamidated (+1)                                         2\n",
      "Deamidated (+1), Deamidated (+1), Deamidated (+1), Deamidated (+1), Deamidated (+1)                              1\n",
      "Oxidation (+16), Oxidation (+16), Deamidated (+1)                                                               16\n",
      "Deamidated (+1), Carbamidomethyl (+57), Carbamidomethyl (+57)                                                    6\n",
      "Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)      2\n",
      "Oxidation (+16), Oxidation (+16), Carbamidomethyl (+57)                                                          1\n",
      "Oxidation (+16), Acetyl (+42), Oxidation (+16)                                                                   1\n",
      "Oxidation (+16), Acetyl (+42), Deamidated (+1)                                                                   1\n",
      "Carbamidomethyl (+57), Deamidated (+1), Carbamidomethyl (+57)                                                    5\n",
      "Oxidation (+16), Deamidated (+1), Carbamidomethyl (+57)                                                          4\n",
      "Deamidated (+1), Carbamidomethyl (+57), Deamidated (+1), Carbamidomethyl (+57)                                   1\n",
      "Acetyl (+42), Carbamidomethyl (+57)                                                                              4\n",
      "Oxidation (+16), Deamidated (+1), Oxidation (+16)                                                                3\n",
      "Deamidated (+1), Oxidation (+16), Carbamidomethyl (+57)                                                          4\n",
      "Carbamidomethyl (+57), Deamidated (+1), Deamidated (+1)                                                          2\n",
      "Carbamidomethyl (+57), Carbamidomethyl (+57), Oxidation (+16)                                                    1\n",
      "Deamidated (+1), Carbamidomethyl (+57), Oxidation (+16)                                                          1\n",
      "Acetyl (+42), Deamidated (+1), Deamidated (+1)                                                                   8\n",
      "Deamidated (+1), Deamidated (+1), Carbamidomethyl (+57)                                                          1\n",
      "Deamidated (+1), Deamidated (+1), Oxidation (+16)                                                                7\n",
      "Carbamidomethyl (+57), Oxidation (+16), Oxidation (+16)                                                          2\n",
      "Oxidation (+16), Deamidated (+1), Deamidated (+1)                                                                1\n",
      "Acetyl (+42), Oxidation (+16)                                                                                    2\n",
      "Oxidation (+16), Carbamidomethyl (+57), Carbamidomethyl (+57)                                                    3\n",
      "Name: count, dtype: int64\n"
     ]
    }
   ],
   "source": [
    "# get count of each distinct occurance of `Modifications identified by spectrum` for NAT\n",
    "nat_modifications = protein_mods_nat['Modifications identified by spectrum']\n",
    "print(f\"The total number of modificaitons for NAT:\", nat_modifications.count())\n",
    "print(f\"\\nThe number of NAT modfications identified by spectrum:\\n\")\n",
    "print(nat_modifications.value_counts(sort = False))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "ef4c6f4f-a89b-4569-b0b7-ff4dd93eff2b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Modifcations found in SPP but not NAT\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Protein accession number</th>\n",
       "      <th>Protein name</th>\n",
       "      <th>Peptide sequence</th>\n",
       "      <th>Peptide identification probability</th>\n",
       "      <th>Modifications identified by spectrum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4956</th>\n",
       "      <td>Q8I3A3</td>\n",
       "      <td>Ubiquitin specific protease, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_0904600 PE=4 SV=1</td>\n",
       "      <td>KNDNIIQNNK</td>\n",
       "      <td>0.911</td>\n",
       "      <td>Deamidated (+1), Deamidated (+1), Deamidated (+1), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12899</th>\n",
       "      <td>Q8I3T8</td>\n",
       "      <td>60S ribosomal protein L12, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_0517000 PE=3 SV=2</td>\n",
       "      <td>EMLGTCNSIGCTVDGK</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Oxidation (+16), Carbamidomethyl (+57), Deamidated (+1), Carbamidomethyl (+57)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13271</th>\n",
       "      <td>Q8II57</td>\n",
       "      <td>Structural maintenance of chromosomes protein 1, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1130700 PE=4 SV=1</td>\n",
       "      <td>QINCKNYLNEKK</td>\n",
       "      <td>0.929</td>\n",
       "      <td>Deamidated (+1), Deamidated (+1), Carbamidomethyl (+57), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13536</th>\n",
       "      <td>Q8IKW5</td>\n",
       "      <td>Elongation factor 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1451100 PE=3 SV=1</td>\n",
       "      <td>YTEQVQDVPCGNTCCLVGVDQYIVK</td>\n",
       "      <td>0.954</td>\n",
       "      <td>Carbamidomethyl (+57), Deamidated (+1), Carbamidomethyl (+57), Carbamidomethyl (+57)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13546</th>\n",
       "      <td>Q8IDN6</td>\n",
       "      <td>Protein transport protein SEC61 subunit alpha OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1346100 PE=3 SV=1</td>\n",
       "      <td>GTLMELGISPIVTSGMVMQLLAGSK</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Oxidation (+16), Oxidation (+16), Oxidation (+16)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16901</th>\n",
       "      <td>Q8I5J4</td>\n",
       "      <td>Uncharacterized protein OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1221900 PE=4 SV=1</td>\n",
       "      <td>KVNKNDEDLNNNSK</td>\n",
       "      <td>0.975</td>\n",
       "      <td>Deamidated (+1), Deamidated (+1), Deamidated (+1), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18003</th>\n",
       "      <td>Q7KQL9</td>\n",
       "      <td>Fructose-bisphosphate aldolase OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=FBPA PE=1 SV=1</td>\n",
       "      <td>AHCTEYMNAPK</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Acetyl (+42), Carbamidomethyl (+57), Oxidation (+16)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18004</th>\n",
       "      <td>Q7KQL9</td>\n",
       "      <td>Fructose-bisphosphate aldolase OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=FBPA PE=1 SV=1</td>\n",
       "      <td>AHCTEYMNAPK</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Acetyl (+42), Carbamidomethyl (+57), Oxidation (+16)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18264</th>\n",
       "      <td>Q8IJT9</td>\n",
       "      <td>Eukaryotic translation initiation factor 2 subunit beta OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1010600 PE=3 SV=1</td>\n",
       "      <td>YITEYVTCQMCK</td>\n",
       "      <td>0.979</td>\n",
       "      <td>Carbamidomethyl (+57), Oxidation (+16), Carbamidomethyl (+57)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18265</th>\n",
       "      <td>Q8IJT9</td>\n",
       "      <td>Eukaryotic translation initiation factor 2 subunit beta OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1010600 PE=3 SV=1</td>\n",
       "      <td>YITEYVTCQMCK</td>\n",
       "      <td>0.989</td>\n",
       "      <td>Carbamidomethyl (+57), Oxidation (+16), Carbamidomethyl (+57)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18587</th>\n",
       "      <td>Q8IIQ3</td>\n",
       "      <td>ES2 protein, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1110900 PE=3 SV=1</td>\n",
       "      <td>EKDRKQQQMIEQGK</td>\n",
       "      <td>0.945</td>\n",
       "      <td>Deamidated (+1), Deamidated (+1), Deamidated (+1), Oxidation (+16)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19047</th>\n",
       "      <td>Q8ILZ7</td>\n",
       "      <td>CUGBP Elav-like family member 2, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1409800 PE=4 SV=1</td>\n",
       "      <td>MSEQCSYQMNSPYNPAPSIPIK</td>\n",
       "      <td>0.921</td>\n",
       "      <td>Oxidation (+16), Carbamidomethyl (+57), Oxidation (+16)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Protein accession number   \n",
       "4956                    Q8I3A3  \\\n",
       "12899                   Q8I3T8   \n",
       "13271                   Q8II57   \n",
       "13536                   Q8IKW5   \n",
       "13546                   Q8IDN6   \n",
       "16901                   Q8I5J4   \n",
       "18003                   Q7KQL9   \n",
       "18004                   Q7KQL9   \n",
       "18264                   Q8IJT9   \n",
       "18265                   Q8IJT9   \n",
       "18587                   Q8IIQ3   \n",
       "19047                   Q8ILZ7   \n",
       "\n",
       "                                                                                                                               Protein name   \n",
       "4956                       Ubiquitin specific protease, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_0904600 PE=4 SV=1  \\\n",
       "12899                        60S ribosomal protein L12, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_0517000 PE=3 SV=2   \n",
       "13271  Structural maintenance of chromosomes protein 1, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1130700 PE=4 SV=1   \n",
       "13536                                        Elongation factor 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1451100 PE=3 SV=1   \n",
       "13546              Protein transport protein SEC61 subunit alpha OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1346100 PE=3 SV=1   \n",
       "16901                                    Uncharacterized protein OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1221900 PE=4 SV=1   \n",
       "18003                                      Fructose-bisphosphate aldolase OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=FBPA PE=1 SV=1   \n",
       "18004                                      Fructose-bisphosphate aldolase OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=FBPA PE=1 SV=1   \n",
       "18264    Eukaryotic translation initiation factor 2 subunit beta OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1010600 PE=3 SV=1   \n",
       "18265    Eukaryotic translation initiation factor 2 subunit beta OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1010600 PE=3 SV=1   \n",
       "18587                                      ES2 protein, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1110900 PE=3 SV=1   \n",
       "19047                  CUGBP Elav-like family member 2, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1409800 PE=4 SV=1   \n",
       "\n",
       "                Peptide sequence  Peptide identification probability   \n",
       "4956                  KNDNIIQNNK                               0.911  \\\n",
       "12899           EMLGTCNSIGCTVDGK                               0.997   \n",
       "13271               QINCKNYLNEKK                               0.929   \n",
       "13536  YTEQVQDVPCGNTCCLVGVDQYIVK                               0.954   \n",
       "13546  GTLMELGISPIVTSGMVMQLLAGSK                               0.997   \n",
       "16901             KVNKNDEDLNNNSK                               0.975   \n",
       "18003                AHCTEYMNAPK                               0.997   \n",
       "18004                AHCTEYMNAPK                               0.997   \n",
       "18264               YITEYVTCQMCK                               0.979   \n",
       "18265               YITEYVTCQMCK                               0.989   \n",
       "18587             EKDRKQQQMIEQGK                               0.945   \n",
       "19047     MSEQCSYQMNSPYNPAPSIPIK                               0.921   \n",
       "\n",
       "                                                       Modifications identified by spectrum  \n",
       "4956                     Deamidated (+1), Deamidated (+1), Deamidated (+1), Deamidated (+1)  \n",
       "12899        Oxidation (+16), Carbamidomethyl (+57), Deamidated (+1), Carbamidomethyl (+57)  \n",
       "13271              Deamidated (+1), Deamidated (+1), Carbamidomethyl (+57), Deamidated (+1)  \n",
       "13536  Carbamidomethyl (+57), Deamidated (+1), Carbamidomethyl (+57), Carbamidomethyl (+57)  \n",
       "13546                                     Oxidation (+16), Oxidation (+16), Oxidation (+16)  \n",
       "16901                    Deamidated (+1), Deamidated (+1), Deamidated (+1), Deamidated (+1)  \n",
       "18003                                  Acetyl (+42), Carbamidomethyl (+57), Oxidation (+16)  \n",
       "18004                                  Acetyl (+42), Carbamidomethyl (+57), Oxidation (+16)  \n",
       "18264                         Carbamidomethyl (+57), Oxidation (+16), Carbamidomethyl (+57)  \n",
       "18265                         Carbamidomethyl (+57), Oxidation (+16), Carbamidomethyl (+57)  \n",
       "18587                    Deamidated (+1), Deamidated (+1), Deamidated (+1), Oxidation (+16)  \n",
       "19047                               Oxidation (+16), Carbamidomethyl (+57), Oxidation (+16)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# find which modifications exist in SPP but not NAT\n",
    "spp_mods_not_in_nat = protein_mods_spp[~protein_mods_spp['Modifications identified by spectrum'].isin(protein_mods_nat['Modifications identified by spectrum'])]\n",
    "print(f\"\\nModifcations found in SPP but not NAT\\n\")\n",
    "# print(spp_mods_not_in_nat.value_counts(sort = False))\n",
    "display(spp_mods_not_in_nat[['Protein accession number', 'Protein name', 'Peptide sequence', 'Peptide identification probability', 'Modifications identified by spectrum']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "8051b092-d832-4243-a143-00081a3a1484",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Modifcations found in NAT but not SPP\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Protein accession number</th>\n",
       "      <th>Protein name</th>\n",
       "      <th>Peptide sequence</th>\n",
       "      <th>Peptide identification probability</th>\n",
       "      <th>Modifications identified by spectrum</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1240</th>\n",
       "      <td>Q8I5Y5</td>\n",
       "      <td>Shewanella-like protein phosphatase 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1206000 PE=4 SV=1</td>\n",
       "      <td>FCVCCYNGPTFNR</td>\n",
       "      <td>0.934</td>\n",
       "      <td>Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1241</th>\n",
       "      <td>Q8I5Y5</td>\n",
       "      <td>Shewanella-like protein phosphatase 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1206000 PE=4 SV=1</td>\n",
       "      <td>FCVCCYNGPTFNR</td>\n",
       "      <td>0.938</td>\n",
       "      <td>Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1242</th>\n",
       "      <td>Q8I5Y5</td>\n",
       "      <td>Shewanella-like protein phosphatase 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1206000 PE=4 SV=1</td>\n",
       "      <td>FCVCCYNGPTFNR</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1243</th>\n",
       "      <td>Q8I5Y5</td>\n",
       "      <td>Shewanella-like protein phosphatase 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1206000 PE=4 SV=1</td>\n",
       "      <td>FCVCCYNGPTFNR</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1282</th>\n",
       "      <td>Q8IIC8</td>\n",
       "      <td>Golgi protein 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1123500 PE=4 SV=1</td>\n",
       "      <td>NKMIDYTNMLQRSK</td>\n",
       "      <td>0.945</td>\n",
       "      <td>Oxidation (+16), Deamidated (+1), Oxidation (+16), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2624</th>\n",
       "      <td>Q8IBQ6</td>\n",
       "      <td>60S ribosomal protein L11a, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_0719600 PE=1 SV=1</td>\n",
       "      <td>EQNVMREIKVNK</td>\n",
       "      <td>0.924</td>\n",
       "      <td>Deamidated (+1), Oxidation (+16), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3010</th>\n",
       "      <td>Q8IL96</td>\n",
       "      <td>N-acetyltransferase, GNAT family, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1437000 PE=4 SV=2</td>\n",
       "      <td>NNNDTCNEQNKDNNNNNNNNNNNNNNQLSK</td>\n",
       "      <td>0.937</td>\n",
       "      <td>Carbamidomethyl (+57), Deamidated (+1), Deamidated (+1), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3275</th>\n",
       "      <td>Q8ID94|YPF12_PLAF7-DECOY</td>\n",
       "      <td>Q8ID94|YPF12_PLAF7-DECOY</td>\n",
       "      <td>IDDPINMSSMVGPVLNNDMNTINNNVTSNKK</td>\n",
       "      <td>0.923</td>\n",
       "      <td>Deamidated (+1), Deamidated (+1), Deamidated (+1), Deamidated (+1), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5719</th>\n",
       "      <td>Q8I3X4</td>\n",
       "      <td>Purine nucleoside phosphorylase OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PNP PE=1 SV=1</td>\n",
       "      <td>FLCVSHGVGSAGCAVCFEELCQNGAK</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5720</th>\n",
       "      <td>Q8I3X4</td>\n",
       "      <td>Purine nucleoside phosphorylase OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PNP PE=1 SV=1</td>\n",
       "      <td>FLCVSHGVGSAGCAVCFEELCQNGAK</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5768</th>\n",
       "      <td>Q8I5Y5</td>\n",
       "      <td>Shewanella-like protein phosphatase 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1206000 PE=4 SV=1</td>\n",
       "      <td>FCVCCYNGPTFNR</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5927</th>\n",
       "      <td>Q8IL96</td>\n",
       "      <td>N-acetyltransferase, GNAT family, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1437000 PE=4 SV=2</td>\n",
       "      <td>NNNDTCNEQNKDNNNNNNNNNNNNNNQLSK</td>\n",
       "      <td>0.923</td>\n",
       "      <td>Carbamidomethyl (+57), Deamidated (+1), Deamidated (+1), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6144</th>\n",
       "      <td>Q8ILX1</td>\n",
       "      <td>Nuclear transport factor 2, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1412300 PE=4 SV=2</td>\n",
       "      <td>DISMMSFENDQCR</td>\n",
       "      <td>0.984</td>\n",
       "      <td>Oxidation (+16), Oxidation (+16), Carbamidomethyl (+57)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6149</th>\n",
       "      <td>Q8ILX1</td>\n",
       "      <td>Nuclear transport factor 2, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1412300 PE=4 SV=2</td>\n",
       "      <td>MDMLNPQFEEIGK</td>\n",
       "      <td>0.969</td>\n",
       "      <td>Oxidation (+16), Acetyl (+42), Oxidation (+16)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6773</th>\n",
       "      <td>Q8IJS7</td>\n",
       "      <td>PRE-binding protein OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1011800 PE=4 SV=1</td>\n",
       "      <td>VTELTSNAHMNGSIQK</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Deamidated (+1), Oxidation (+16), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8170</th>\n",
       "      <td>Q8ILS9|Q8ILS9_PLAF7-DECOY</td>\n",
       "      <td>Q8ILS9|Q8ILS9_PLAF7-DECOY</td>\n",
       "      <td>NKDMPINK</td>\n",
       "      <td>0.940</td>\n",
       "      <td>Deamidated (+1), Oxidation (+16), Deamidated (+1)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10489</th>\n",
       "      <td>Q8IM23</td>\n",
       "      <td>rRNA 2'-O-methyltransferase fibrillarin, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1407100 PE=3 SV=1</td>\n",
       "      <td>LGACLMGGVGNMPIKPGSK</td>\n",
       "      <td>0.989</td>\n",
       "      <td>Carbamidomethyl (+57), Oxidation (+16), Oxidation (+16)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10490</th>\n",
       "      <td>Q8IM23</td>\n",
       "      <td>rRNA 2'-O-methyltransferase fibrillarin, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1407100 PE=3 SV=1</td>\n",
       "      <td>LGACLMGGVGNMPIKPGSK</td>\n",
       "      <td>0.997</td>\n",
       "      <td>Carbamidomethyl (+57), Oxidation (+16), Oxidation (+16)</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Protein accession number   \n",
       "1240                      Q8I5Y5  \\\n",
       "1241                      Q8I5Y5   \n",
       "1242                      Q8I5Y5   \n",
       "1243                      Q8I5Y5   \n",
       "1282                      Q8IIC8   \n",
       "2624                      Q8IBQ6   \n",
       "3010                      Q8IL96   \n",
       "3275    Q8ID94|YPF12_PLAF7-DECOY   \n",
       "5719                      Q8I3X4   \n",
       "5720                      Q8I3X4   \n",
       "5768                      Q8I5Y5   \n",
       "5927                      Q8IL96   \n",
       "6144                      Q8ILX1   \n",
       "6149                      Q8ILX1   \n",
       "6773                      Q8IJS7   \n",
       "8170   Q8ILS9|Q8ILS9_PLAF7-DECOY   \n",
       "10489                     Q8IM23   \n",
       "10490                     Q8IM23   \n",
       "\n",
       "                                                                                                                       Protein name   \n",
       "1240               Shewanella-like protein phosphatase 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1206000 PE=4 SV=1  \\\n",
       "1241               Shewanella-like protein phosphatase 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1206000 PE=4 SV=1   \n",
       "1242               Shewanella-like protein phosphatase 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1206000 PE=4 SV=1   \n",
       "1243               Shewanella-like protein phosphatase 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1206000 PE=4 SV=1   \n",
       "1282                                     Golgi protein 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1123500 PE=4 SV=1   \n",
       "2624                60S ribosomal protein L11a, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_0719600 PE=1 SV=1   \n",
       "3010          N-acetyltransferase, GNAT family, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1437000 PE=4 SV=2   \n",
       "3275                                                                                                       Q8ID94|YPF12_PLAF7-DECOY   \n",
       "5719                               Purine nucleoside phosphorylase OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PNP PE=1 SV=1   \n",
       "5720                               Purine nucleoside phosphorylase OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PNP PE=1 SV=1   \n",
       "5768               Shewanella-like protein phosphatase 2 OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1206000 PE=4 SV=1   \n",
       "5927          N-acetyltransferase, GNAT family, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1437000 PE=4 SV=2   \n",
       "6144                Nuclear transport factor 2, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1412300 PE=4 SV=2   \n",
       "6149                Nuclear transport factor 2, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1412300 PE=4 SV=2   \n",
       "6773                                 PRE-binding protein OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1011800 PE=4 SV=1   \n",
       "8170                                                                                                      Q8ILS9|Q8ILS9_PLAF7-DECOY   \n",
       "10489  rRNA 2'-O-methyltransferase fibrillarin, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1407100 PE=3 SV=1   \n",
       "10490  rRNA 2'-O-methyltransferase fibrillarin, putative OS=Plasmodium falciparum (isolate 3D7) OX=36329 GN=PF3D7_1407100 PE=3 SV=1   \n",
       "\n",
       "                      Peptide sequence  Peptide identification probability   \n",
       "1240                     FCVCCYNGPTFNR                               0.934  \\\n",
       "1241                     FCVCCYNGPTFNR                               0.938   \n",
       "1242                     FCVCCYNGPTFNR                               0.997   \n",
       "1243                     FCVCCYNGPTFNR                               0.997   \n",
       "1282                    NKMIDYTNMLQRSK                               0.945   \n",
       "2624                      EQNVMREIKVNK                               0.924   \n",
       "3010    NNNDTCNEQNKDNNNNNNNNNNNNNNQLSK                               0.937   \n",
       "3275   IDDPINMSSMVGPVLNNDMNTINNNVTSNKK                               0.923   \n",
       "5719        FLCVSHGVGSAGCAVCFEELCQNGAK                               0.997   \n",
       "5720        FLCVSHGVGSAGCAVCFEELCQNGAK                               0.997   \n",
       "5768                     FCVCCYNGPTFNR                               0.997   \n",
       "5927    NNNDTCNEQNKDNNNNNNNNNNNNNNQLSK                               0.923   \n",
       "6144                     DISMMSFENDQCR                               0.984   \n",
       "6149                     MDMLNPQFEEIGK                               0.969   \n",
       "6773                  VTELTSNAHMNGSIQK                               0.997   \n",
       "8170                          NKDMPINK                               0.940   \n",
       "10489              LGACLMGGVGNMPIKPGSK                               0.989   \n",
       "10490              LGACLMGGVGNMPIKPGSK                               0.997   \n",
       "\n",
       "                                                                              Modifications identified by spectrum  \n",
       "1240                          Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)  \n",
       "1241                          Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)  \n",
       "1242                          Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)  \n",
       "1243                          Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)  \n",
       "1282                                            Oxidation (+16), Deamidated (+1), Oxidation (+16), Deamidated (+1)  \n",
       "2624                                                             Deamidated (+1), Oxidation (+16), Deamidated (+1)  \n",
       "3010                                      Carbamidomethyl (+57), Deamidated (+1), Deamidated (+1), Deamidated (+1)  \n",
       "3275                           Deamidated (+1), Deamidated (+1), Deamidated (+1), Deamidated (+1), Deamidated (+1)  \n",
       "5719   Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)  \n",
       "5720   Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)  \n",
       "5768                          Carbamidomethyl (+57), Carbamidomethyl (+57), Carbamidomethyl (+57), Deamidated (+1)  \n",
       "5927                                      Carbamidomethyl (+57), Deamidated (+1), Deamidated (+1), Deamidated (+1)  \n",
       "6144                                                       Oxidation (+16), Oxidation (+16), Carbamidomethyl (+57)  \n",
       "6149                                                                Oxidation (+16), Acetyl (+42), Oxidation (+16)  \n",
       "6773                                                             Deamidated (+1), Oxidation (+16), Deamidated (+1)  \n",
       "8170                                                             Deamidated (+1), Oxidation (+16), Deamidated (+1)  \n",
       "10489                                                      Carbamidomethyl (+57), Oxidation (+16), Oxidation (+16)  \n",
       "10490                                                      Carbamidomethyl (+57), Oxidation (+16), Oxidation (+16)  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# find which modifications exist in NAT but not SPP\n",
    "nat_mods_not_in_spp = protein_mods_nat[~protein_mods_nat['Modifications identified by spectrum'].isin(protein_mods_spp['Modifications identified by spectrum'])]\n",
    "print(f\"\\nModifcations found in NAT but not SPP\\n\")\n",
    "# print(spp_mods_not_in_nat.value_counts(sort = False))\n",
    "display(nat_mods_not_in_spp[['Protein accession number', 'Protein name', 'Peptide sequence', 'Peptide identification probability', 'Modifications identified by spectrum']])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1a345e0d-aacb-4532-bd35-a3c1aeccd7d9",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
